runtime/tests/checker/type_inference_test.go

```diff
@@ -1094,10 +1094,13 @@ func TestCheckDictionarySupertypeInference(t *testing.T) {
 				},
 			},
 			{
-				name:              "no supertype for inner keys",
-				code:              `let x = {0: {10: 1, 20: 2}, 1: {"one": 1, "two": 2}}`,
-				expectedKeyType:   sema.IntType,
-				expectedValueType: sema.AnyStructType,
+				name:            "no supertype for inner keys",
+				code:            `let x = {0: {10: 1, 20: 2}, 1: {"one": 1, "two": 2}}`,
+				expectedKeyType: sema.IntType,
+				expectedValueType: &sema.DictionaryType{
+					KeyType:   sema.HashableStructType,
+					ValueType: sema.IntType,
+				},
```

Comment:
Does this look okay?

Now the leastCommonSuperType of `{10: 1, 20: 2}` and `{"one": 1, "two": 2}` is `{HashableStruct:Int}` rather than `AnyStruct`.

This is because now the common super type of `Int` and `String` is `HashableStruct` instead of `AnyStruct`. So the if condition

```
if !IsSubType(keySuperType, HashableStructType) {
  return commonSuperTypeOfHeterogeneousTypes(types)\
}
``` 

in `commonSuperTypeOfDictionaries` in type_tags.go returns `false` as `HashableStructType` is a sub-type of `HashableStructType`. Earlier it used to go inside the `if` condition and return `AnyStruct`.

Reply:
Yes, `HashableStruct` being inferred as the least-common supertype for `Int` and `String` looks correct, and so does `{HashableStruct: Int}` for `{Int: Int}` and `{String: Int}` 👍

Reply:
👍